lib/google.js

```diff
@@ -0,0 +1,222 @@
+'use strict';
+
+/* global JuttleAdapterAPI */
+let _ = require('underscore');
+let logger = JuttleAdapterAPI.getLogger('google-analytics');
+let JuttleMoment = JuttleAdapterAPI.types.JuttleMoment;
+let Promise = require('bluebird');
+let google = require('googleapis');
+let analytics = google.analytics('v3');
+
+let auth, authorized, accountId;
+
+function init(config) {
+    auth = new google.auth.JWT(
+        config.service_account.client_email,
+        null,
+        config.service_account.private_key,
+        ['https://www.googleapis.com/auth/analytics'],
+        null
+    );
+    Promise.promisifyAll(auth);
+    Promise.promisifyAll(analytics.data.ga);
+    Promise.promisifyAll(analytics.management.accounts);
+    Promise.promisifyAll(analytics.management.webproperties);
+    Promise.promisifyAll(analytics.management.profiles);
+}
+
+function authorize() {
+    if (! authorized) {
+        authorized = auth.authorizeAsync()
+        .then(() => {
+            return analytics.management.accounts.listAsync({auth: auth});
+        })
+        .then((accounts) => {
+            accountId = accounts.items[0].id;
+        });
+    }
+    return authorized;
+}
+
+// given 1 or more strings, prefix each with "ga:" and return a comma separated
+// string.
+function gaString(vals) {
+    if (! (vals instanceof Array)) {
+        vals = [vals];
+    }
+    return vals.map((val) => { return 'ga:' + val; }).join(', ');
+}
+
+function toJuttle(results, extras) {
+    let points = [];
+    let columns = results.columnHeaders;
+    let data = results.rows;
+
+    if (!data) {
+        return [];
+    }
+
+    for (let row = 0; row < data.length; ++row) {
+        let pt = extras ? _.clone(extras) : {};
+        for (let col = 0; col < columns.length; ++col) {
+            let spec = columns[col];
+            let name = spec.name.substr(3);
+            let value = data[row][col];
+
+            if (spec.dataType === 'INTEGER') {
+                value = parseInt(value, 10);
+            } else if (spec.dataType === 'TIME') {
+                value = parseFloat(value);
+            }
+            switch (name) {
```

Comment:
What's going on here? Seems like some of these set the time and others add something to it so if they come in the wrong order we'll have some problems.